test/e2e/pods.go

```diff
@@ -34,11 +34,22 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+// createNamespaceIfDoesNotExist ensures that the namespace with specified name exists, or returns an error
+func createNamespaceIfDoesNotExist(c *client.Client, name string) (*api.Namespace, error) {
```

Comment:
No major disagreement, but I am really tired of running 50 minute tests against GCE.

This was good enough for me now.  I think these tests should really move to the common framework longer term.